pkg/api/norman/customization/clusterregistrationtokens/formatter.go

```diff
@@ -75,6 +78,22 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 			token,
 			ca,
 			getWindowsPrefixPathArg(cluster.Spec.RancherKubernetesEngineConfig))
+	} else {
+		// refresh nodeCommand if agent image or env vars changed
+		cluster, err := f.ClusterLister.Get("", clusterID)
```

Comment:
I think we should just do an API call here, if the intention is to be updating things getting the most recent version seems like a safer option and we really aren't going to hit this code much.